### PR TITLE
feat: workflow and script to sort readme

### PR DESCRIPTION
**Pull request by bearsyankees**

### CODE REVIEW

**Comment by bearsyankees** [comment]
fixes #3361

**Comment by bearsyankees** [comment]
@yan-michael @rushilsrivastava thoughts? :)

**Review by rushilsrivastava — CHANGES_REQUESTED** [review]
Hey! Thanks for the contribution! I think instead of making another action, it's best to expand upon the code @matttreed wrote and modify the `sortListings` function [here](https://github.com/SimplifyJobs/Summer2024-Internships/blob/dev/.github/scripts/util.py#L132-L159).

**Comment by bearsyankees** [comment]
Good call. That line should do it :) @rushilsrivastava

**Review by Kartikay77 — APPROVED** [review]

File: .github/scripts/util.py [hunk 1]
```diff
@@ -141,13 +142,9 @@ def sortListings(listings):
 
     listings.sort(
         key=lambda x: (
-            datetime(
-                datetime.fromtimestamp(x["date_posted"]).year,
-                datetime.fromtimestamp(x["date_posted"]).month,
-                datetime.fromtimestamp(x["date_posted"]).day
-            ),
+            x["active"],  # Active listings first
+            datetime.fromtimestamp(x["date_posted"]),
```
Comment (rushilsrivastava) [inline .github/scripts/util.py]: This doesn't strip out hours, minutes, seconds, and microseconds from the timestamp, so it won't group postings from the same company on the same day together.
```suggestion
            datetime(
                datetime.fromtimestamp(x["date_posted"]).year,
                datetime.fromtimestamp(x["date_posted"]).month,
                datetime.fromtimestamp(x["date_posted"]).day
            ),
```

**Review by rushilsrivastava — CHANGES_REQUESTED** [review]

**Pull request merged**